pkg/controllers/build/container_controller.go

```diff
@@ -95,24 +107,47 @@ func (r *ContainerReconciler) reconcile(ctx context.Context, log logr.Logger, co
 		}
 		return ctrl.Result{}, err
 	}
+	container.Status.TargetImage = targetImageRef.Name()
+
+	latestImage, err := r.resolveDigestReference(ctx, log, targetImageRef, container)
+	if err != nil {
+		container.Status.MarkImageInvalid(err.Error())
+		return ctrl.Result{}, err
+	}
+
 	container.Status.MarkImageResolved()
-	container.Status.TargetImage = targetImage
 
-	// TODO resolve to a digest
-	container.Status.LatestImage = container.Status.TargetImage
+	container.Status.LatestImage = latestImage
 
 	container.Status.ObservedGeneration = container.Generation
 
-	return ctrl.Result{}, nil
+	return ctrl.Result{
+		Requeue:      true,
```

Comment:
Requeue is implicitly true when RequeueAfter is set

https://godoc.org/sigs.k8s.io/controller-runtime/pkg/reconcile#Result